### PR TITLE
refactor: Notifications rework

### LINKED ISSUE #14725: Notifications rework
**Is your feature request related to a problem? Please describe.**
Currently each component that need to show notification has a different logic for it.
In general, the component tracks a Collection in mongodb. When something is changed it needs to compare with last version and decide if it is time to show a notification or not.

**Describe the solution you'd like**
Refactor all current notifications to use the new pattern (#14703) where all notifications are sent from akka-apps.

#### user join/left
- [x] UserJoinedMeeting (PR #14737 - akka-apps only)
- [x] UserLeftMeeting (PR #14737 - akka-apps only)

#### breakout rooms
- [x] BreakoutRoomEnded (PR #14728)
- [x] UserMessageToAllBreakouts (PR #14728)
- [x] BreakoutDurationUpdated - viewers (PR #14703)
- [x] BreakoutDurationUpdated - moderator (PR #14703)

#### recording
- [x] RecordingStarted (PR #14737 - akka-apps only)
- [x] RecordingStopped (PR #14737 - akka-apps only)

#### guest user
- [x] NewGuestUser (PR #14737 - akka-apps only)

#### lock settings
- [x] DisableCamLockSetting (PR #14737 - akka-apps only)
- [x] EnableCamLockSetting (PR #14737 - akka-apps only)
- [x] DisableMicLockSetting (PR #14737 - akka-apps only)
- [x] EnableMicLockSetting (PR #14737 - akka-apps only)
- [x] DisablePrivateChatLockSetting (PR #14737 - akka-apps only)
- [x] EnablePrivateChatLockSetting (PR #14737 - akka-apps only)
- [x] DisablePublicChatLockSetting (PR #14737 - akka-apps only)
- [x] EnablePublicChatLockSetting (PR #14737 - akka-apps only)
- [x] DisableNotesLockSetting (PR #14737 - akka-apps only)
- [x] EnableNotesLockSetting (PR #14737 - akka-apps only)
- [x] HideUserListLockSetting (PR #14737 - akka-apps only)
- [x] ShowUserListLockSetting (PR #14737 - akka-apps only)
- [x] OnlyModeratorWebcamDisabledLockSetting (PR #14737 - akka-apps only)
- [x] OnlyModeratorWebcamEnabledLockSetting (PR #14737 - akka-apps only)

#### mute/unmute
- [x] MeetingMuted (PR #14737 - akka-apps only)
- [x] MeetingUnmuted (PR #14737 - akka-apps only)

#### polls
- [x] HasPublishedPoll (PR #14737 - akka-apps only)

#### promote/demote
- [x] UserPromotedToModerator (PR #14737 - akka-apps only)
- [x] UserDemotedToViewer (PR #14737 - akka-apps only)

Thanks @JoVictorNunes for the [notification_list.pdf](https://github.com/bigbluebutton/bigbluebutton/files/8399382/notification_list.pdf)

### PR DESCRIPTION
**Pull request by ramonlsouza**
### What does this PR do?

Refactor all current notifications to use the new pattern, where all notifications are sent from akka-apps.

Includes #14728, #14751 and #14737 + resolves conflicts and minor adjustments.

### Closes Issue(s)
Closes #14725

### CODE REVIEW

File: bigbluebutton-html5/imports/ui/components/notifications/container.jsx [hunk 1]
```diff
@@ -1,12 +1,26 @@
...
 import { notify } from '/imports/ui/services/notification';
 import { withTracker } from 'meteor/react-meteor-data';
+import WaitingUsersAlertService from '/imports/ui/components/waiting-users/alert/service';
+import UserService from '/imports/ui/components/user-list/service';
 
-export default withTracker(() => {
+export default injectIntl(withTracker(({ intl }) => {
   NotificationsCollection.find({}).observe({
     added: (obj) => {
+      if (obj.messageId === 'app.userList.guest.pendingGuestAlert') {
```
Comment (antobinary) [inline bigbluebutton-html5/imports/ui/components/notifications/container.jsx:12]: Should the NotificationsCollection.find filter by meetingId? An edge case would be receiving breakout1 annotations in the main room or vice versa

Comment (ramonlsouza) [reply bigbluebutton-html5/imports/ui/components/notifications/container.jsx:12]: this is handled by the publisher (https://github.com/bigbluebutton/bigbluebutton/blob/develop/bigbluebutton-html5/imports/api/meetings/server/publishers.js#L143-L154)

File: akka-bbb-apps/src/main/scala/org/bigbluebutton/core/apps/users/RegisterUserReqMsgHdlr.scala [hunk 2]
```diff
@@ -60,6 +60,16 @@ trait RegisterUserReqMsgHdlr {
         val guest = GuestWaiting(regUser.id, regUser.name, regUser.role, regUser.guest, regUser.avatarURL, regUser.authed, regUser.registeredOn)
         addGuestToWaitingForApproval(guest, liveMeeting.guestsWaiting)
         notifyModeratorsOfGuestWaiting(Vector(guest), liveMeeting.users2x, liveMeeting.props.meetingProp.intId)
+        val notifyEvent = MsgBuilder.buildNotifyRoleInMeetingEvtMsg(
+          "MODERATOR",
```
Comment (gustavotrott) [inline akka-bbb-apps/src/main/scala/org/bigbluebutton/core/apps/users/RegisterUserReqMsgHdlr.scala:64]:
```suggestion
          Roles.MODERATOR_ROLE,

```

**Review by gustavotrott — APPROVED** [review]
LGTM!

**Pull request merged**